### PR TITLE
Clang tidy deflaking / experiment in ci

### LINKED ISSUE #546: clang-tidy flake
In https://github.com/envoyproxy/nighthawk/pull/533 I observed a first occurrency of a non-reproducible clang-tidy flake.
Since this CI task has proven to be quite stable over time, it's worth tracking further occurrences of this here.

If this turns out problematic, the first potential suspect I would investigate is the last Envoy update, which at this time is 8fcf416d1e31bf1f6f2e8688c86f24495aef87b7

### PR DESCRIPTION
**Pull request by oschaaf**
At some point clang-tidy runs broke in CI, and this couldn't be reproduced
locally. The after a while local runs would exit with a failure seemingly mid-
run.  This PR addresses both these things.

- Update the CircleCI SHA of the envoy build docker image.
- Sync the script we use to run Envoy's build docker image.
- Make clang-tidy runs use libstdc++ to avoid standard library
  headers from producing errors.
- Sync run_clang_tidy.sh
- Update MAINTAINERS.md to mention we should keep more things in sync.

Fixes #546 

Signed-off-by: Otto van der Schaaf <ovanders@redhat.com>

---

- [x] Prerequisite: https://github.com/envoyproxy/nighthawk/pull/630
- [x] Rebase after merge of ^^

### CODE REVIEW

**Comment by oschaaf** [comment]
I'm a bit at a loss here, it's almost as if the clang tidy replacement binary is crashing or some such.

**Comment by mum4k** [comment]
@qqustc please review and assign back to me once done.

**Comment by oschaaf** [comment]
Unfortunately the clang-tidy check now fails after merging master here.
But I suspect one important part of this change may only get effective after we merge: the update of the circleci configuration.

**Review by qqustc — APPROVED** [review]
LGTM, reassign back to Jakub for review

File: .circleci/config.yml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 references:
-  envoy-build-image: &envoy-build-image # November 10th, 2020
-    envoyproxy/envoy-build-ubuntu:19a268cfe3d12625380e7c61d2467c8779b58b56
+  envoy-build-image: &envoy-build-image # Februari 14th, 2021
```
Comment (mum4k) [inline .circleci/config.yml:2]: (nit) Typo: February

**Review by mum4k — CHANGES_REQUESTED** [review]
@oschaaf how do we want to proceed with regards to the failing clang-tidy on this PR?

File: .circleci/config.yml (hunk 1 above)
Comment (oschaaf) [reply .circleci/config.yml]: Fixed in e122081. Let's stick to English :-)

**Comment by oschaaf** [comment]
> @oschaaf how do we want to proceed with regards to the failing clang-tidy on this PR?

I found out that it's possible to ssh into the CI env, I'm trying to figure out what's up. I'm stumped that clang_tidy passed after updating the docker image sha in the circleci config, and subsequently started failing again after merging main into this branch.

**Comment by oschaaf** [comment]
Allright, the PR resolves a few problems, but we it looks like we would sometimes silently get OOM killed. Aargh. In any case, let's respin CI  to give the integration tests a chance to resolve.
/retest

**Review by mum4k — APPROVED** [review]

**Pull request merged**